### DISCUSSION TITLE
Can Kong Gateway be used in front of a monolith?

### DISCUSSION THREAD

**Discussion by khawarizmus** [Kong/kong#7184] [Help] [answered]
Can Kong Gateway be used in front of a monolith offloading authentication and ACL and many other things?

**Comment by mheap** [comment, marked as answer]
Hey @gimyboya! It can indeed. Kong Gateway can route to any upstream service you have, including handling auth + ACL for your monolith.

If at any point in the future you want to start decomposing that monolith into microservices, Kong can help route the incoming requests to the correct service so that your customers aren't impacted
